packages/astro/src/runtime/server/render/component.ts

```diff
@@ -421,6 +421,9 @@ function renderAstroComponent(
 	slots: any = {}
 ): RenderInstance {
 	const instance = createAstroComponentInstance(result, displayName, Component, props, slots);
+	// Initialize the instance early so that child components with head propagation
+	// are able to register themselves.
+	instance.init(result);
```

Comment:
Interesting, I thought we were already calling it in `bufferHeadContent`:

https://github.com/withastro/astro/blob/0fa483283e54c94f173838cd558dc0dbdd11e699/packages/astro/src/runtime/server/render/astro/render.ts#L152-L167

which the propagators is added during component instance creation too:

https://github.com/withastro/astro/blob/0fa483283e54c94f173838cd558dc0dbdd11e699/packages/astro/src/runtime/server/render/astro/instance.ts#L77-L89

Could it be that the component we're rendering failed the `isAPropagatingComponent()` check?

Reply:
Ah actually that's a good point. The parent should have marked as `'in-tree'`. I'll check on this.

Reply:
Ok, turns out that we were not bubbling up `propagation` that came through the compiler. In that case we don't need to early call to init.